packages/core/src/state-management/repository.ts

```diff
@@ -318,18 +303,46 @@ export class Repository {
    * Request anchor for the latest stream state
    */
   async anchor(state$: RunningState, opts: AnchorOpts): Promise<Subscription> {
-    return this.stateManager.anchor(state$, opts)
+    if (!this.anchorService) {
+      throw new Error(`Anchor requested for stream ${state$.id} but anchoring is disabled`)
+    }
+    if (state$.value.anchorStatus == AnchorStatus.ANCHORED) {
+      return
+    }
+
+    const carFile = await this.#deps.anchorRequestCarBuilder.build(state$.id, state$.tip)
+    const genesisCID = state$.value.log[0].cid
+    const genesisCommit = carFile.get(genesisCID)
+    await this.anchorRequestStore.save(state$.id, {
+      cid: state$.tip,
+      timestamp: Date.now(),
+      genesis: genesisCommit,
+    })
+
+    const anchorStatus$ = await this.anchorService.requestAnchor(
+      carFile,
+      opts.waitForAnchorConfirmation
+    )
+    return this._internals.processAnchorResponse(state$, anchorStatus$)
   }
 
   /**
    * Apply options relating to authoring a new commit
    *
    * @param state$ - Running State
    * @param opts - Initialization options (request anchor, publish to pubsub, etc.)
+   * @param opType - If we load, create or update a stream
    * @private
    */
   async applyWriteOpts(state$: RunningState, opts: CreateOpts | UpdateOpts, opType: OperationType) {
-    await this.stateManager.applyWriteOpts(state$, opts, opType)
+    const anchor = (opts as any).anchor
+    const publish = (opts as any).publish
```

Comment:
Do `opts` really need to be cast to `any`? Seems the properties should be property typed, no?

Reply:
yep, seems fine to remove.  will do in next PR